### PR TITLE
Unsubscribe from interior data on various events

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Refactored dependency injection for RC plugin commands
Added InteriorDataManager class
Moved managing subscription on events to InteriorDataManager

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/rc_command_params.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#ifndef RC_COMMAND_PARAMS_H
```
Comment (LitvinenkoIra) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/rc_command_params.h]: @LuxoftAKutsan please fix header guards

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/rc_command_params.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+namespace rc_rpc_plugin {
+
+class ResourceAllocationManager;
+class InteriorDataCache;
+class InteriorDataManager;
+
+struct RCCommandParams {
+  //    const application_manager::commands::MessageSharedPtr& message_;
```
Comment (LitvinenkoIra) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/rc_command_params.h]: @LuxoftAKutsan commented code?

Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/rc_command_params.h]: @LitvinenkoIra 859658637

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/rc_command_params.h (hunk 1 above)
Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/rc_command_params.h]: @LitvinenkoIra 859658637

**Pull request merged**